smtp/smtp.js

```diff
@@ -274,70 +396,39 @@ class SMTP extends EventEmitter {
 		});
 	}
 
+	/**
+	 * @param {function(...*): void} callback function to call after response
+	 * @returns {void}
+	 */
 	starttls(callback) {
 		const response = (err, msg) => {
 			if (err) {
 				err.message += ' while establishing a starttls session';
 				caller(callback, err);
 			} else {
-				// support new API
-				if (tls.TLSSocket) {
-					const secured_socket = new tls.TLSSocket(this.sock, {
-						secureContext: tls.createSecureContext
-							? tls.createSecureContext(this.tls)
-							: crypto.createCredentials(this.tls),
-						isServer: false, // older versions of node (0.12), do not default to false properly...
-					});
-
-					secured_socket.on('error', err => {
-						this.close(true);
-						caller(callback, err);
-					});
+				const secureContext = createSecureContext({});
```

Comment:
`createSecureContext` will always exist on node instances with access to `TLSSocket`

Reply:
i had this for older node support, but i think this was for <4 and we aren't supporting that anymore.

Reply:
this is where `this.tls` should be passed in if it's an object; re-added in 26e838ae32f984180bf07a1ab9dd574c34062d42